cpu/esp32/Makefile.features

```diff
@@ -1,5 +1,8 @@
 # Features that are provided by the CPU independent on the board
 FEATURES_PROVIDED += cpp
+# Features that are provided by CPU independent on the board
```

Comment:
There is a problem with this comment and the alphabetical order. `cpp` should be after `esp32` I think.

Reply:
they are in different lists.

Reply:
`cpp` is in the same FEATURES_PROVIDED list as arch_xx. See arm7_common features list.

Reply:
moved arch features up to CPU defined features